docs/docs/reference/implicit-by-name-parameters.md

```diff
@@ -0,0 +1,42 @@
+# Implicit By-Name Parameters
+
+Call-by-name implicit parameters can be used to avoid a divergent implicit expansion.
+
+```scala
+implicit def serializeOption[T](implicit ev: => Serializable[T]): Serializable[Option[T]] =
+  new Serializable[Option[T]] {
+    def write(x: Option[T], out: DataOutputStream) = t match {
+      case Some(x) => ev.write(x)
+      case None =>
+    }
+  }
+
+  val s = serializeOption[Option[Int]]
+  s.write(Some(33))
+  s.write(None)
+```
+As is the case for a normal by-name parameter, the argument for the implicit parameter `ev`
+is evaluated on demand. In the example above, if the option value `x` is `None`, it is
+not evaluated at all.
+
+The synthesized argument for an implicit parameter is backed by a lazy
+val, which means that the parameter is evaluated at most once. The
```

Comment:
No, it's complicated. The lazy val is an internal implementation detail of the synthesized argument. It is not generated at all unless the synthesis is recursive. The parameter itself is by-name, i.e. the argument computation can be invoked multiple times.